packages/foam-vscode/src/core/janitor/generateHeadings.test.ts

```diff
@@ -16,11 +17,11 @@ describe('generateHeadings', () => {
   const findBySlug = (slug: string): Resource => {
     return _workspace
       .list()
-      .find(res => URI.getBasename(res.uri) === slug) as Resource;
+      .find(res => getName(res.uri.path) === slug) as Resource;
```

Comment:
I agree with the rationale, but this is the reason why all the functions in `core/utils/path.ts` manipulate uri/posix paths. So `uri.path` is an object manipulated by functions in `core/utils/path.ts`. Since those in `core/utils/path.ts` are functions over strings, I'm not able to leverage the type system. Having a `Path` object has some drawbacks. Having everything encapsulated inside `URI` has some drawbacks. I've commented about this below and also done a lot of experimentation. So, although I agree with the not having to deal with filesystem specific paths part, my preference is *against* having the path API replicated inside URI, and also *against* having a path abstraction represented as a `Path` class. Instead, I proposed to use the new platform-agnostic functions on the uri.path string.

As a halfway solution I'm adding some path functions to URI, but I'm not very enthusiastic about it.